ksqldb-functional-tests/src/test/resources/query-validation-tests/collect-list.json

```diff
@@ -108,6 +108,76 @@
       ]
     },
+    {
+      "name": "collect_list timestamp map",
+      "format": ["JSON", "PROTOBUF"],
+      "statements": [
+        "CREATE STREAM TEST (ID BIGINT KEY, NAME varchar, VALUE map<varchar, timestamp>) WITH (kafka_topic='test_topic', value_format='{FORMAT}');",
+        "CREATE TABLE S2 as SELECT ID, collect_list(value['key1']) AS collected FROM test group by id;"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "key": 0, "value": {"name": "zero", "value": {"key1":10, "key2":15}}},
+        {"topic": "test_topic", "key": 0, "value": {"name": "zero", "value": {"key1":20, "key2":25}}},
+        {"topic": "test_topic", "key": 0, "value": {"name": "zero", "value": {"key1":30, "key2":35}}}
+      ],
+      "outputs": [
+        {"topic": "S2", "key": 0, "value": {"COLLECTED":[10]}},
+        {"topic": "S2", "key": 0, "value": {"COLLECTED":[10,20]}},
+        {"topic": "S2", "key": 0, "value": {"COLLECTED":[10,20,30]}}
+      ]
+    },
+    {
+      "name": "collect_list time map",
+      "format": ["JSON", "PROTOBUF"],
+      "statements": [
+        "CREATE STREAM TEST (ID BIGINT KEY, NAME varchar, VALUE map<varchar, time>) WITH (kafka_topic='test_topic', value_format='{FORMAT}');",
+        "CREATE TABLE S2 as SELECT ID, collect_list(value['key1']) AS collected FROM test group by id;"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "key": 0, "value": {"name": "zero", "value": {"key1":10, "key2":15}}},
+        {"topic": "test_topic", "key": 0, "value": {"name": "zero", "value": {"key1":20, "key2":25}}},
+        {"topic": "test_topic", "key": 0, "value": {"name": "zero", "value": {"key1":30, "key2":35}}}
+      ],
+      "outputs": [
+        {"topic": "S2", "key": 0, "value": {"COLLECTED":[10]}},
+        {"topic": "S2", "key": 0, "value": {"COLLECTED":[10,20]}},
+        {"topic": "S2", "key": 0, "value": {"COLLECTED":[10,20,30]}}
+      ]
+    },
+    {
+      "name": "collect_list date map",
+      "format": ["JSON", "PROTOBUF"],
+      "statements": [
+        "CREATE STREAM TEST (ID BIGINT KEY, NAME varchar, VALUE map<varchar, date>) WITH (kafka_topic='test_topic', value_format='{FORMAT}');",
+        "CREATE TABLE S2 as SELECT ID, collect_list(value['key1']) AS collected FROM test group by id;"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "key": 0, "value": {"name": "zero", "value": {"key1":10, "key2":15}}},
+        {"topic": "test_topic", "key": 0, "value": {"name": "zero", "value": {"key1":20, "key2":25}}},
+        {"topic": "test_topic", "key": 0, "value": {"name": "zero", "value": {"key1":30, "key2":35}}}
+      ],
+      "outputs": [
+        {"topic": "S2", "key": 0, "value": {"COLLECTED":[10]}},
+        {"topic": "S2", "key": 0, "value": {"COLLECTED":[10,20]}},
+        {"topic": "S2", "key": 0, "value": {"COLLECTED":[10,20,30]}}
+      ]
+    },
+    {
+      "name": "collect_list bytes map",
+      "format": ["JSON"],
```

Comment:
Out of curiosity, does the PROTOBUF format not support the BYTES type?

Reply:
It does, it just requires specifying the schema to the test, which I was too lazy to do.